internal/pkg/v2/ui/http/api/common/batch/batch.go

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+ * Copyright 2020 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package batch
+
+import (
+	"net/http"
+
+	"github.com/edgexfoundry/edgex-go/internal/pkg/v2/ui/common"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/v2/ui/http/api"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/v2/ui/http/handle"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/v2/ui/http/routing"
+
+	"github.com/edgexfoundry/go-mod-bootstrap/di"
+
+	"github.com/gorilla/mux"
+)
+
+const (
+	Endpoint = "/api/batch"
```

Comment:
In swagger document, batch is still under /api/v2, and I suggest we still give batch api a version in the path for the following reason:

1. It might be a problem if users can mixed different versions of request, especially we will remove v1 API in the future.
2. After getting more user feedback, we might still want to modify the design of batch request in the future, especially about the error handling.  Without the version in the path, we might be easy to encounter backward compatibility issue, although the current design looks good so far.

Reply:
> In swagger document, batch is still under /api/v2

No, it isn't.  See https://github.com/edgexfoundry/edgex-go/blob/7ca3df59c01c7f56d3c83a840c5995cdb7718c20/api/openapi/v2/core-command.yaml#L302.  See also https://github.com/edgexfoundry/edgex-go/pull/2309#discussion_r368939010.

> I suggest we still give batch API a version in the path

The batch endpoint is our interface for alternate transport implementations -- think of asynchronous messaging.  The intent is for it to be able to take various different major-release versions of request DTOs going forward; i.e. a v2 use-case request and a v3 use-case request within the same transport request.

Reply:
I disagree with versioning the batch endpoint.  Explanation follows.

> In swagger document, batch is still under /api/v2

No, it isn't.  See https://github.com/edgexfoundry/edgex-go/blob/7ca3df59c01c7f56d3c83a840c5995cdb7718c20/api/openapi/v2/core-command.yaml#L302.  See also https://github.com/edgexfoundry/edgex-go/pull/2309#discussion_r368939010.

> I suggest we still give batch API a version in the path

The batch endpoint is our interface for alternate transport implementations -- think of asynchronous messaging.  The intent is for it to be able to take various different major-release versions of request DTOs going forward; i.e. a v2 use-case request and a v3 use-case request within the same transport request.  As such, it makes no sense to version the URL.

Currently, the envelopes associated with the batch endpoint are not versioned.  This is intentional.  Consider that they currently include an implicit version field and number (that doesn't correspond to the API's version number).  If it becomes necessary to modify the envelope structure in the future, the intent is to add an optional explicit version field (and codify the existing implied version number) to support the changes.

Reply:
Got it.  Thanks for the elaboration.